src/context.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+interface PortalProviderProps {
+  children?: React.ReactNode;
+}
+
+export interface PortalValue {
+  setPortal: (name: string, children: React.ReactNode) => void;
+  getPortal: (name: string) => React.ReactNode | undefined;
+}
+
+export const PortalContext = React.createContext<PortalValue | null>(null);
+
+export function PortalProvider(props: PortalProviderProps): JSX.Element {
+  const [portals, setPortals] = React.useState<Map<string, React.ReactNode>>(
+    new Map()
+  );
+
+  const setPortal = React.useCallback(
+    (name: string, children: React.ReactNode): void => {
+      const newPortals = new Map(portals);
+      newPortals.set(name, children);
+      setPortals(newPortals);
```

Comment:
If we keep `Map`, what do you think about this style (from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Cloning_and_merging_Maps):

```suggestion
      setPortals(new Map([...portals, [name, children]]));
```

Reply:
6ac67d5